shell/platform/embedder/embedder.cc

```diff
@@ -624,6 +732,59 @@ static sk_sp<SkSurface> MakeSkSurfaceFromBackingStore(
 #endif
 }
 
+static sk_sp<SkSurface> MakeSkSurfaceFromBackingStore(
+    GrDirectContext* context,
+    const FlutterBackingStoreConfig& config,
+    const FlutterVulkanBackingStore* vulkan) {
+#ifdef SHELL_ENABLE_VULKAN
+  if (!vulkan->image) {
+    FML_LOG(ERROR) << "Embedder supplied null Vulkan image.";
+    return nullptr;
+  }
+  GrVkImageInfo image_info = {
+      .fImage = reinterpret_cast<VkImage>(vulkan->image),
+      .fImageTiling = VK_IMAGE_TILING_OPTIMAL,
+      .fImageLayout = VK_IMAGE_LAYOUT_UNDEFINED,
+      .fFormat = static_cast<VkFormat>(vulkan->format),
+      .fImageUsageFlags = VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT |
```

Comment:
We aren't going to be transferring to and from this, or, sampling in a shader from this image. I think just the color attachment usage bit ought to be fine. All backing stores in the embedder API are render targets. These textures are also unlikely (and probably can't if they are obtained via a swapchain extension) to be allocated with these usage flags in the embedders. If we incorrectly tag these, embedder might trip validation.

Reply:
Ah yes, I'm glad you asked about this because I need to add a comment here. You're right that `VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT` is the only usage flag required for these VkImages passed into the embedder. But note that no Vulkan resources are being created here, and this field (as well as a number of other fields) are literally just used for Skia to validate things.

SkSurface has an API design here that's incompatible with our use case. Rather than selectively disabling corresponding functionality on SkSurface when flags are missing, SkSurface instead always forces all 4 flags be provided at construction time in order to presume that all possible SkSurface ops are safe.

*Checks notes...*

```
GrVkGpu.cpp: 1204
VK_IMAGE_USAGE_TRANSFER_SRC_BIT
VK_IMAGE_USAGE_TRANSFER_DST_BIT

GrVkGpu.cpp: 1247
VK_IMAGE_USAGE_SAMPLED_BIT

GrVkGpu.cpp: 1258
VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT
```